nixos/modules/config/system-path.nix

```diff
@@ -120,18 +120,17 @@ in
         "/share/vim-plugins"
       ];
 
-    environment.outputsToLink = [ "bin" "lib" "out" ];
-
     system.path = pkgs.buildEnv {
       name = "system-path";
       paths =
-        lib.filter (drv: drv != null && drv != (drv.dev or null))
-          (lib.concatMap (drv:
-            [ drv ] ++ map (outputName: drv.${outputName}.outPath or null) config.environment.outputsToLink)
-           config.environment.systemPackages);
-      inherit (config.environment) pathsToLink;
+        # The default output probably shouldn't be globally configurable.
+        # Services and users should specify them explicitly unless they want this default.
+        map (p: if p.outputUnspecified or false then p.bin or p.out or p else p)
```

Comment:
This should probably be factored out into a separate function, e.g. `getOutputsToInstall` or something like that, so it can be re-used in places like the handling of `systemd.services.<unit>.path`.

This should probably also be overridable via a package attribute, e.g.
```
meta.outputsToInstall = [ "bin" "man" ];
```
which would also be respected by `nix-env`.